spec.md

```diff
@@ -487,6 +490,92 @@ The tags MUST be in lexical order.
 
 When using the `last` query parameter, the `n` parameter is OPTIONAL.
 
+##### Listing Referrers
+
+*Note: this feature was added in distibution-spec 1.1.
+Clients should see [client-implementation.md](client-implementation.md) for more details on implementing this.
+Registries should see [upgrading.md](upgrading.md) before enabling this.*
+
+To fetch the list of referrers, perform a `GET` request to a path in the following format: `/v2/<name>/referrers/<reference>` <sup>[end-12a](#endpoints)</sup>.
+
+`<name>` is the namespace of the repository.
+Assuming a repository is found, this request MUST return a `200 OK` response code.
+
+Upon success, the response MUST be a JSON body with an Index containing a list of descriptors.
+Each descriptor is of an Image or Artifact manifest in the same `<name>` namespace with a `refers` field that specifies the value of `<reference>`.
+The descriptors MUST include an `artifactType` field that is set to the value of `artifactType` for an Artifact manifest if present, or the configuration descriptor's `mediaType` for an Image manifest.
+The descriptors MUST include annotations from the manifest of the Image or Artifact.
+If a query results in no matching referrers, an empty manifest list MUST be returned.
+If `<ref>` does not exist, a registry MAY return an empty list.
+
+```json
+{
```

Comment:
noting that this example is not included in the image spec PR...

Reply:
In this context, it's similar to having the tag listing output displayed in the section above. While we use the Index spec, the part we want to capture is how that Index is generated by the server and what content must be included in the descriptors and annotations.

Reply:
right.. if json has comments would be nice to note in the json that this was a generated image.index .. not a pushed image.index  

I'm having difficulty figuring out use patterns with this PR for adding artifact manifests pre image.index manifest push, and post image.index manifest push (both in the case where the referrers api is fully supported/registry and client are both using proposed new fields). I get that In the pre push pattern would the referrers api on the registry only pull from the image.index manifests.. or is there some expectation that manifests with a refers would be otherwise indexed in another manner than with image.index manifests in some sort of refers db?  Is there an expectation for using both a db and image.index for artifacts? I see some implication of such below in upgrading.md 's MUST language but I'm not seeing a requirement that the image.index be inclusive making me wonder what the practice will be for image.index push/pull with artifact manifests included, vs this referrers api with filter and paging support... I can make the guess that it's index+additional image and artifact manifests linked by refers.. and there might be good reason for needing that to reduce updates to the image.index.

Reply:
I'm not sure I understand the full question, but when a registry supports the API, we're assuming the registry will maintain an internal DB of refers. There would be no referrers tag schema pointing to an Index that can be separately pulled in that situation (for new objects at least). All updates would be done in that DB and the Index is dynamically generated by the API. Whether the client created tag schema Index manifests are deleted after the API is enabled is a choice for the registry.

We document what manifests registries must include in that DB below because scanning all existing manifests for the refers field may be excessive for large registries.

Reply:
can I push an image.index with artifacts and image manifests that include refers ..  

can I push an image.index with artifacts that do not include refers..

based on those answers:

is there an implied refer for the image.index when it includes an artifact without a refer reference

is an artifact included in the image.index required to include a refer to one of the manifest image/image index/other artifact via refer

Reply:
appears the answers could all be yes, moving this question/issue to the punch list... to be covered in later pr(s)